### ISSUE TITLE
Expose the short name for a vocabulary

### ISSUE THREAD

**Issue by kouralex** [NatLibFi/Skosmos#1089]
As @osma said in https://github.com/NatLibFi/Skosmos/pull/1084#issuecomment-717238010, short names for vocabularies can be language specific. However, it turns out the related code parts (listed below)

https://github.com/NatLibFi/Skosmos/blob/5a7175ed01fdba9108dbf219680f2d4d4f9257fc/model/VocabularyConfig.php#L106-L114
https://github.com/NatLibFi/Skosmos/blob/5a7175ed01fdba9108dbf219680f2d4d4f9257fc/model/BaseConfig.php#L39-L64
https://github.com/NatLibFi/Skosmos/blob/5a7175ed01fdba9108dbf219680f2d4d4f9257fc/model/Concept.php#L207-L210
https://github.com/NatLibFi/Skosmos/blob/5a7175ed01fdba9108dbf219680f2d4d4f9257fc/model/Vocabulary.php#L633-L635

are not applicable for fetching the short name in another language than the UI language (which, I guess, is the usual use case).

However, now, if one wants to display the vocabulary short name in a plugin in the content language, it simply can not be done in a straightforward way (I know that it is possible to make parametrized plugins but it would feel a bit awkward way to convey vocabulary configuration related information via parameters to the plugin).

It could be possible to change the signature of the getShortName() method in VocabularyConfig.php so that the required information could be fetched and used directly in twig templates and exposed to JavaScript (via the normal https://github.com/NatLibFi/Skosmos/blob/master/view/scripts.twig procedure).

There is just one problem here, the current implementation of getLiteral() in BaseConfig.php limits the values returned (may be arbitrary in case of a non-existing literal language value) as default value is only returned in case no value is found for the property in question (this could be solved by following the language order of the vocabulary).

@osma do you think this is a plausible solution?

**Comment by osma** [comment]
I agree that the short name of the vocabulary needs to be exported to the plugins, probably as a javascript variable like other similar information (vocabulary id, concept URI etc).

> However, now, if one wants to display the vocabulary short name in a plugin in the content language, it simply can not be done in a straightforward way (I know that it is possible to make parametrized plugins but it would feel a bit awkward way to convey vocabulary configuration related information via parameters to the plugin).

I don't see a need for expressing the vocabulary name in the content language. For one, it may not be defined at all (consider e.g. Lexvo which has many content languages) since the vocabulary names and short names are usually defined only in the UI languages in the configuration file.
